python/extensions/aproc/proc/ingest/drivers/ingest_driver.py

```diff
@@ -80,40 +89,52 @@ def identify_assets(self, url: str) -> list[Asset]:
         ...
 
     @abstractmethod
-    def fetch_assets(self, url: str, resources: list[Asset]) -> list[Asset]:
+    def fetch_assets(self, url: str, assets: list[Asset]) -> list[Asset]:
         """Copy or download the assets locally
 
         Args:
             url (str): archive's url
-            resources (list[Asset]): list of assets to be fetched
+            assets (list[Asset]): list of assets to be fetched
 
         Returns:
             list[Asset]: list of fetched assets. Assets must have a valid name, href and roles. Assets href must be existing local files.
         """
         ...
 
     @abstractmethod
-    def transform_assets(self, url: str, resources: list[Asset]) -> list[Asset]:
+    def transform_assets(self, url: str, assets: list[Asset]) -> list[Asset]:
         """Transform the assets, if necessary
 
         Args:
             url (str): archive's url
-            resources (list[Asset]): list of assets to be transformed
+            assets (list[Asset]): list of assets to be transformed
 
         Returns:
             list[Asset]: list of transformed assets. Assets must have a valid name, href and roles. Assets href must be existing local files.
         """
         ...
 
     @abstractmethod
-    def to_item(self, url: str, resources: list[Asset]) -> Item:
+    def to_item(self, url: str, assets: list[Asset]) -> Item:
         """Analyse an archive assets to create an item
 
         Args:
             url (str): archive's url
-            resources (list[Asset]): list of assets. Assets must have a valid name, href and roles. Assets href must be existing local files.
+            assets (list[Asset]): list of assets. Assets must have a valid name, href and roles. Assets href must be existing local files.
 
         Returns:
             Item: the item. An item must have a valid id and valid assets.
         """
         ...
+
+    @abstractmethod
+    def __check_path__(self, path: str) -> bool:
```

Comment:
this is indeed often called by the impl, but this was not part of the interface, also it is an internal method (__). I do not understand the goal of putting it in the interface